File: web/src/api/admin/gost_user_config.js
import request from "../../request/index.js";

const baseUrl = '/v1/admin/gost/user/config'

export const apiAdminGostUserConfigPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiAdminGostUserConfigList = (data) => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
        data
    })
}

export const apiAdminGostUserConfigNodeList = (data) => {
    return request.request({
        url: `${baseUrl}/node/list`,
        method: 'POST',
        data
    })
}

export const apiAdminGostUserConfigCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiAdminGostUserConfigUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}


export const apiAdminGostUserConfigDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}

File: server/router/admin/gost_user_config.go
package admin

import (
	"github.com/gin-gonic/gin"
	"server/controller/admin/gost_user_config"
	"server/global"
	"server/router/middleware"
)

func InitGostUserConfig(group *gin.RouterGroup) {
	g := group.Group("gost/user/config", middleware.Auth(global.Jwt), middleware.AuthAdmin())
	g.POST("create", gost_user_config.Create)
	g.POST("delete", gost_user_config.Delete)
	g.POST("update", gost_user_config.Update)
	g.POST("page", gost_user_config.Page)
	g.POST("list", gost_user_config.List)
	g.POST("node/list", gost_user_config.NodeList)
}
